dev/archery/archery/utils/logger.py

```diff
@@ -27,3 +29,21 @@ def __init__(self, quiet=False):
 
 
 ctx = LoggingContext()
+
+in_github_actions = (os.environ.get("GITHUB_ACTIONS") == "true")
+
+
+@contextmanager
+def group(name):
+    """
+    Group outputs in the given with block.
+
+    This does nothing in non GitHub Actions environment for now.
+    """
+    if in_github_actions:
+        print(f"::group::{name}")
```

Comment:
Hmm. This isn't outputted immediately...
Do we need to `flush` or something...?

Reply:
Wow! I missed it.

Can we move the `log` and `_Printer` to `archery.utils.printer` or something from `archery.integration.util`?
I want to use it in `archery docker` too.

Reply:
Then I would suggest something like:
```python

@contextmanager
def group(name, log=print):
    """
    Group outputs in the given with block.
    This does nothing in non GitHub Actions environment for now.
    """
    if in_github_actions:
        log(f"::group::{name}")
    try:
        yield
    finally:
        if in_github_actions:
            log("::endgroup::")
```

Then we can keep `log` integration-specific.

Reply:
I've used this style.